org.opent2t.sample.lamp.superpopular/com.philips.hue.bulb/js/thingTranslator.js

```diff
@@ -40,56 +16,133 @@ function validateArgumentType(arg, argName, expectedType) {
     }
 }
 
+// Helper method to convert the device schema to the translator schema.
+function deviceSchemaToTranslatorSchema(deviceSchema) {
+
+    var deviceState = deviceSchema.state;
+    var result = {
+        id:  'lights.' + deviceId,
+        rt: 'org.opent2t.sample.lamp.superpopular'
+    };
+    
+    if(typeof deviceState != 'undefined'){
+        result.n = deviceSchema['name'] ;
+        result.power = { value: deviceState['on'] };
+        result.dim = { dimmingSetting: deviceState['bri'], range: [1, 254]};
+    } else {
+        for (var i = 0; i < deviceSchema.length; i++){
+            var changeResult = deviceSchema[i].success;
+            if(typeof changeResult != 'undefined'){
+                var key = Object.keys(changeResult)[0];
+                var parseKey = key.split("/");
+                var attribute = parseKey[parseKey.length-1];
+                if( attribute == 'on'){
+                    result.power = { value: changeResult[key] };
+                } else if ( attribute == 'bri'){
+                    result.dim = { dimmingSetting: changeResult[key], range: [1, 254]};
+                } else if ( attribute == 'name'){
+                    result.n = changeResult[key];
+                }
+            }
+        }
+    }
+    return result;
+}
+
+// Helper method to convert the translator schema to the device schema.
+function translatorSchemaToDeviceSchema(translatorSchema) {
 
-var device;
-var deviceProps;
-var api;
+    // build the object with desired state
+    var result = { };
 
-// module exports, implementing the schema
-module.exports = {
+    if (translatorSchema.n !== undefined) {
+        result['name'] = translatorSchema.n;
+    }
 
-    initDevice: function (dev) {
+    if (translatorSchema.power!== undefined) {
+        result['on'] = translatorSchema.power.value;
+    }
+
+    if (translatorSchema.dim!== undefined) {
+        result['bri'] = translatorSchema.dim.dimmingSetting;
+    }
+
+    return result;
+}
+
+var deviceId;
+var deviceType = 'lights';
+var hueHelper;
+
+// This translator class implements the 'org.opent2t.sample.light.superpopular' schema.
+class Translator {
+
+    constructor(device) {
```

Comment:
The Nest and Wink translators have moved away from having *helper classes, and instead use a concept of "hubs" even where a physical hub doesn't exist (like with Nest).  The "hub" is then passed to the translator in the constructor.